### PR TITLE
URLEncoding of query string parameters

### LINKED ISSUE #442: Get-VSTeamVariableGroup by name throws error 404 with TFS 2020 u1.1
Steps to reproduce
------------------

```powershell
Set-VSTeamAccount "http://tfs.acme.com:8080/tfs/MyCollection/" -UseWindowsAuthentication
Get-VSTeamVariableGroup -ProjectName MyProject -Name MyGroup
```

Expected behavior
-----------------
A group record

Actual behavior
---------------

Error message with a large HTML along the lines of "error 404"

Retrieval by ID works.

If you set the project by `Set-VSTeamDefaultProject`, same result.

Environment data
----------------

OS

- [ ] macOS
- [x] Windows
- [ ] Linux

Server

- [ ] TFS 2017
- [ ] TFS 2018
- [x] Azure DevOps Server
- [ ] Azure DevOps Service

```powershell
Billing                     :
Build                       : 3.0
Core                        : 3.0
DistributedTask             : 3.0-preview
DistributedTaskReleased     :
ExtensionsManagement        : 3.0-preview
Git                         : 3.0
Graph                       :
HierarchyQuery              :
MemberEntitlementManagement :
Packaging                   : 3.0-preview
Pipelines                   :
Policy                      : 3.0
Processes                   :
Release                     : 3.0-preview
ServiceEndpoints            : 3.0-preview
TaskGroups                  : 3.0-preview
Tfvc                        : 3.0
VariableGroups              :
Version                     : TFS2017
```

```
Name                           Value
----                           -----
PSVersion                      5.1.18362.1801
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.18362.1801
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### PR DESCRIPTION
**Pull request by sevaa**
# PR Summary

Refactor to get rid of crude and incorrect _appendQueryString. Query string parameter values that can potentially be strings should be URLEncoded. Also, _appendQueryString was assuming that the URL already had a query string and produced invalid URLs if the API version was blank.

Fixes #442, #517

## PR Checklist

- [ ] [Write Help](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-help)
- [x] [Write Unit Test](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-unit-test)
- [x] [Update CHANGELOG.md](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#update-changelogmd)

### CODE REVIEW

**Comment by sevaa** [comment]
@SebastianSchuetze  The tests are failing for me on PowerShell 5 because `-Depth` was not a valid parameter on `ConvertFrom-Json` until PS 6.2. Should I:
 - Get rid of -Depth - it's an optimization at best, not a hard requirement
 - Introduce a condition
 - Not test on PS5 at all
?

**Comment by SebastianSchuetze** [comment]
Do you mean they fail locally on your machine or are you talking about the test failing in the pipeline. In the future PS5 was planned to not be supported anymore with VSTeam. That would be Version 8.0.0. As for now I would suggest the one that makes the shift as easy as possible to version 8 but still working with 7.x.

Which would be the best option from your side?

**Comment by sevaa** [comment]
Failing on my machine. There is no PS5 build in the pipeline.

I'd prefer to remove the -Depth. This will break nothing.

PS5.1 is still the version that ships with Windows 10/11 and Windows Server. PS 6-7 is an optional install.

**Comment by sevaa** [comment]
@SebastianSchuetze Ready,

File: Source/Public/Get-VSTeamUserEntitlement.ps1 [hunk 1]
```diff
@@ -94,21 +95,34 @@ function Get-VSTeamUserEntitlement {
...
-            $listurl += _appendQueryString -name "select" -value ($select -join ",")
+            $qs.Add("top", $top)
+            $qs.Add("skip", $skip)
+            if($select) {
+               $qs.Add("select", $select -join ",")
+            }
 
             # Call the REST API
+            # TODO: will break with blank API version
+            $listurl += "&" + $qs.ToString()
```
Comment (SebastianSchuetze) [inline Source/Public/Get-VSTeamUserEntitlement.ps1]: What do you think has to be done here or where exactly?

Comment (sevaa) [reply Source/Public/Get-VSTeamUserEntitlement.ps1]: For correctness? Same thing I did elsewhere:
```
if($listurl.IndexOf('?') -gt 0) {
    $listurl += "&" + $qs.ToString()
}
else {
    $listurl += "?" + $qs.ToString()
}
```

**Pull request merged**